# This fixes an infinite loop in the case of:
#
#	:set leftright
#	:set number
#
# See PR 28687.  Patch has been submitted to maintainer.

diff -ru nvi-1.81.4.orig/vi/vs_refresh.c nvi-1.81.4/vi/vs_refresh.c
--- nvi-1.81.4.orig/vi/vs_refresh.c	Wed Jul 19 10:05:20 2000
+++ nvi-1.81.4/vi/vs_refresh.c	Sat Jul  7 16:11:51 2001
@@ -573,7 +573,7 @@
 		 * for the number option offset.
 		 */
 		cnt = vs_columns(sp, NULL, LNO, &CNO, NULL);
-		if (O_ISSET(sp, O_NUMBER))
+		if (O_ISSET(sp, O_NUMBER) && cnt >= O_NUMBER_LENGTH)
 			cnt -= O_NUMBER_LENGTH;
 
 		/* Adjust the window towards the beginning of the line. */
